docs/sanity-migration/09-neon-integration.md

```diff
@@ -0,0 +1,487 @@
+Contest entries for prize drawings.
+
+| Column | Type | Description |
+|--------|------|-------------|
+| `id` | UUID | Primary key |
+| `user_id` | UUID | Foreign key to users (nullable) |
+| `contest_sanity_id` | VARCHAR(255) | Sanity contest document ID |
+| `first_name` | VARCHAR(64) | Entry first name |
+| `last_name` | VARCHAR(64) | Entry last name |
+| `email` | VARCHAR(255) | Entry email |
+| `phone` | VARCHAR(20) | Entry phone |
+| `hometown` | VARCHAR(64) | Entry hometown |
+| `newsletter_opt_in` | BOOLEAN | Newsletter opt-in |
+| `display` | BOOLEAN | Include in winner pool |
+| `submitted_at` | TIMESTAMP | Entry submission time |
+
+**Unique constraint:** One entry per email per contest.
+
+### Helper Functions
+
+#### `pick_winner(p_contest_id VARCHAR)`
+Randomly selects a winner from eligible contest entries.
+
+**Returns:** `first_name`, `last_name`, `email`, `phone`
+
+**Usage:**
+```sql
+SELECT * FROM pick_winner('top11-2025-11-20');
+```
+
+#### `get_vote_counts(p_contest_id VARCHAR)`
+Aggregates vote counts and weighted scores for a contest.
+
+**Returns:** `sanity_option_id`, `year_end_poll_category_sanity_id`, `total_votes`, `weighted_score`
+
+**Weighted scoring:** For ranked votes, score = (12 - top_11_rank). #1 choice = 11 points, #11 choice = 1 point.
+
+**Usage:**
+```sql
+SELECT * FROM get_vote_counts('top11-2025-11-20') ORDER BY weighted_score DESC;
+```
+
+#### `get_write_ins(p_contest_id VARCHAR)`
+Lists write-in submissions with counts.
+
+**Returns:** `write_in_value`, `category`, `count`
+
+**Usage:**
+```sql
+SELECT * FROM get_write_ins('yep-2025') WHERE count >= 5;
+```
+
+---
+
+## Sanity Schema Reference
+
+### Contest Types
+
+#### Top 11 Contest
+Weekly song countdown contest.
+
+**Document ID format:** `top11-{year}-{month}-{date}` (e.g., `top11-2025-11-20`)
+
+**Key fields:**
+- `date` - Contest date
+- `songs` - Array of Song references
+- `votingOpensAt`, `votingClosesAt` - Voting window
+- `summary` - Portable text for description, links, and embeds
+- `status` - Contest status
+
+**Fixed values:**
+- Max selections: Always 11 (enforced in application)
+- Write-ins: Always allowed
+
+#### Top 11 Result
+Published weekly results.
+
+**Key fields:**
+- `contest` - Reference to top11Contest
+- `placements` - Array of rank/song/artist objects
+- `publishedAt` - Publication timestamp
+
+#### Year End Poll
+Annual poll configuration.
+
+**Document ID format:** `yep-{year}`
+
+**Key fields:**
+- `year` - Poll year
+- `votingOpensAt`, `votingClosesAt` - Voting window
+- `status` - 'draft' | 'open' | 'closed' | 'archived'
+
+#### Year End Poll Category
+Individual poll category (e.g., "Best Album").
+
+**Document ID format:** `yep-{year}-{categorySlug}`
+
+**Key fields:**
+- `poll` - Reference to yearEndPoll
+- `categoryType` - 'songs' | 'records' | 'artists' | 'concerts' | 'freeform'
+- `options` - Array (structure varies by categoryType)
+- `maxSelections` - How many votes per user
+
+#### MRM Tournament
```

Comment:
I'll continue to review this later

Reply:
This is a lot to review at once. Can we split the three Neon contest models into three PRs? One for Top11 (probably ready to land) and another for Year End Poll (close to ready) and Modern Rock Madness (just getting started)